Home.cpp

```diff
@@ -0,0 +1,60 @@
+#include "Home.h"
+#include "Pets.h"
+#include <iostream>
+
+Home::Home(Pets *group) {
+    this -> group = group;
+}
+Pets& Home::getPetByIndex(int i)  {
+    return (Pets&)this -> group[i];
+}
+void Home::printAllPets()  {
+    for (int i =0;  i<3; i++) {
+        getPetByIndex(i).printPet();
+    }
+}
+void Home::findFriends(){
+    for (int i=0; i<3; i++) {
+        for (int j = 0; j < 3; j++) {
+            if(i==j){
+                continue;
+            }
+            if(this ->getPetByIndex(i).get_age()- this->getPetByIndex(j).get_age()<=1){
+                std::cout<<getPetByIndex(i).get_name()<<"and"<<getPetByIndex(j).get_name()<<"are friend!\n";
+            }
+        }
+    }
+}
+int Home :: isPolite() {
+    for (int i = 0; i < 3; i++) {
```

Comment:
3 is the magic number and is used in code multiple times. Ude #define to declare it or make it constant